app/routes.py

```diff
@@ -1,2 +1,231 @@
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f"Task {task.task_id} \"{task.title}\" successfully deleted"})
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+
+    task = Task.query.get(task_id)
+    if not task is None:
+        task.completed_at = datetime.utcnow()
+        db.session.commit()
+        if task.title == "My Beautiful Task":
+            
+            client=slack.WebClient(token=os.getenv('SLACK_TOKEN'))
+            client.chat_postMessage(
+                    channel="slack-api-test-channel", 
+                    text=f"Someone just completed the task {task.title}"
+                )
+
+        return make_response({"task": { "id": task.task_id,
+                                                "title": task.title,
+                                                "description": task.description,
+                                                "is_complete": bool(task.completed_at)
+                                                }}, 200)
+        
+    else:
+        return make_response("",404)
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if not task is None:
+        task.completed_at = None
+        db.session.commit()
+
+        return make_response({"task": { "id": task.task_id,
+                                                "title": task.title,
+                                                "description": task.description,
+                                                "is_complete": bool(task.completed_at) #added bool
+                                                }}, 200)
+    else:
+        return make_response("",404)
+
+
+##### goals ######
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST","GET"])
+def handle_goals():
+    if request.method=="POST":
+        request_body = request.get_json()
+        if "title" not in request_body.keys():
+            return make_response({"details": "Invalid data"}, 400) 
+        else:
+            new_goal = Goal(title=request_body["title"])
+            db.session.add(new_goal)
+            db.session.commit()
+            
+            return make_response({"goal": {
+                    "id": new_goal.goal_id,
+                    "title": new_goal.title
+                }}, 201)
+
+    elif request.method=="GET":
+        title_query = request.args.get('sort')
+        if title_query=='desc':
+            goals = Goal.query.order_by(Goal.title.desc()).all()
+        elif title_query=='asc':
+            goals = Goal.query.order_by(Goal.title).all()
+        else:
+            goals=Goal.query.all()
+        goals_response=[]
+        for goal in goals:
+            goals_response.append({
+                    "id": goal.goal_id,
+                    "title": goal.title
```

Comment:
you could make this a helper function or move it to the goal model as an instance method